packages/fxa-payments-server/src/store/amplitude-middleware.ts

```diff
@@ -0,0 +1,187 @@
+        );
+        break;
+      case 'createSubscription_REJECTED':
+        logAmplitudeEvent(
+          eventGroupNames.createSubscription,
+          eventTypeNames.fail,
+          {
+            ...getPlanPropsFromAction(action),
+            ...getFailureReasonFromAction(action),
+          }
+        );
+        break;
+
+      case 'updatePaymentMounted':
+        logAmplitudeEvent(
+          eventGroupNames.updatePayment,
+          eventTypeNames.view,
+          getPlanPropsFromAction(action)
+        );
+        break;
+      case 'updatePaymentEngaged':
+        logAmplitudeEvent(
+          eventGroupNames.updatePayment,
+          eventTypeNames.engage,
+          getPlanPropsFromAction(action)
+        );
+        break;
+      case 'updatePayment_PENDING':
+        logAmplitudeEvent(
+          eventGroupNames.updatePayment,
+          eventTypeNames.submit,
+          getPlanPropsFromAction(action)
+        );
+        break;
+      case 'updatePayment_FULFILLED':
+        logAmplitudeEvent(
+          eventGroupNames.updatePayment,
+          eventTypeNames.success,
+          getPlanPropsFromAction(action)
+        );
+        logAmplitudeEvent(
+          eventGroupNames.updatePayment,
+          eventTypeNames.complete,
+          getPlanPropsFromAction(action)
+        );
+        break;
+      case 'updatePayment_REJECTED':
+        logAmplitudeEvent(eventGroupNames.updatePayment, eventTypeNames.fail, {
+          ...getPlanPropsFromAction(action),
+          ...getFailureReasonFromAction(action),
+        });
+        break;
+
+      case 'cancelSubscriptionMounted':
+        logAmplitudeEvent(
+          eventGroupNames.cancelSubscription,
+          eventTypeNames.view,
+          getPlanPropsFromAction(action)
+        );
+        break;
+      case 'cancelSubscriptionEngaged':
+        logAmplitudeEvent(
+          eventGroupNames.cancelSubscription,
+          eventTypeNames.engage,
+          getPlanPropsFromAction(action)
+        );
+        break;
+      case 'cancelSubscription_PENDING':
+        logAmplitudeEvent(
+          eventGroupNames.cancelSubscription,
+          eventTypeNames.submit,
+          getPlanPropsFromAction(action)
+        );
+        break;
+      case 'cancelSubscription_FULFILLED':
+        logAmplitudeEvent(
+          eventGroupNames.cancelSubscription,
+          eventTypeNames.success,
+          getPlanPropsFromAction(action)
+        );
+        logAmplitudeEvent(
+          eventGroupNames.cancelSubscription,
+          eventTypeNames.complete,
+          getPlanPropsFromAction(action)
+        );
+        break;
+      case 'cancelSubscription_REJECTED':
+        logAmplitudeEvent(
+          eventGroupNames.cancelSubscription,
+          eventTypeNames.fail,
+          {
+            ...getPlanPropsFromAction(action),
+            ...getFailureReasonFromAction(action),
+          }
+        );
+        break;
+    }
+  } catch (e) {
+    // NOOP
```

Comment:
Again, I wish we could do something to let ourselves know when metrics start breaking. Could we use raven/sentry or something? Maybe better for a followup bug than to block on this now.